crates/anyrender_skia/src/scene.rs

```diff
@@ -0,0 +1,675 @@
+    ) {
+        self.inner.save();
+        self.inner
+            .set_matrix(&kurbo_affine_to_skia_matrix(transform).into());
+
+        let mut paint = anyrender_brush_to_skia_paint(brush.into(), brush_transform);
+        paint.set_style(PaintStyle::Fill);
+        paint.set_anti_alias(true);
+
+        draw_kurbo_shape_to_skia_canvas(self.inner, shape, &paint);
+
+        self.inner.restore();
+    }
+
+    fn draw_glyphs<'a, 's: 'a>(
+        &'s mut self,
+        font: &'a peniko::FontData,
+        font_size: f32,
+        hint: bool,
+        normalized_coords: &'a [anyrender::NormalizedCoord],
+        style: impl Into<peniko::StyleRef<'a>>,
+        brush: impl Into<anyrender::PaintRef<'a>>,
+        brush_alpha: f32,
+        transform: kurbo::Affine,
+        glyph_transform: Option<kurbo::Affine>,
+        glyphs: impl Iterator<Item = anyrender::Glyph>,
+    ) {
+        self.inner.save();
+        self.inner
+            .set_matrix(&kurbo_affine_to_skia_matrix(transform).into());
+
+        if let Some(affine) = glyph_transform {
+            self.inner.concat(&kurbo_affine_to_skia_matrix(affine));
+        }
+
+        let mut paint = anyrender_brush_to_skia_paint(brush.into(), None);
+        apply_peniko_style_to_skia_paint(style.into(), &mut paint);
+        paint.set_alpha_f(brush_alpha);
+        paint.set_anti_alias(true);
+
+        let font_key = (font.data.id(), font.index);
+
+        if !self.typeface_cache.contains_key(&font_key) {
+            let Some(typeface) = self
+                .font_mgr
+                .new_from_data(&font.data.data(), font.index as usize)
+            else {
+                let tf = Typeface::make_deserialize(font.data.data(), None);
+                eprintln!(
+                    "WARNING: failed to load font {} {} {}",
+                    font_key.0,
+                    font_key.1,
+                    tf.is_some()
+                );
+                return;
+            };
+            self.typeface_cache.insert(font_key, typeface);
+        }
+
+        let original_typeface = self.typeface_cache.get(&font_key).unwrap();
+        let mut normalized_typeface: Option<Typeface> = None;
+
+        fn f2dot14_to_f32(raw_value: i16) -> f32 {
+            let int = (raw_value >> 14) as f32;
+            let fract = (raw_value & !(!0 << 14)) as f32 / (1 << 14) as f32;
+            int + fract
+        }
+
+        if !normalized_coords.is_empty() {
+            let axes = original_typeface
+                .variation_design_parameters()
+                .unwrap_or(vec![]);
+            if !axes.is_empty() {
+                let coordinates: Vec<Coordinate> = axes
+                    .iter()
+                    .zip(normalized_coords.iter().map(|c| f2dot14_to_f32(*c)))
+                    .filter(|(_, value)| *value != 0.0)
+                    .map(|(axis, factor)| {
+                        let value = axis.min + ((axis.max - axis.min) * factor);
```

Comment:
`factor` is actually in the range `[-1, 1]` with negative values mapped to `[axis.min, axis.default]` and positive values mapped to `[axis.default, axis.max]` so this should be something like:
```rust
if factor < 0.0 {
    lerp(axis.default, axis.min, -factor)
} else {
    lerp(axis.default, axis.max, factor)
}
```